tests/components/prometheus/test_init.py

```diff
@@ -87,6 +87,71 @@
 PROMETHEUS_PATH = "homeassistant.components.prometheus"
 
 
+@dataclass(frozen=True)
+class EntityTestMetricInfo:
+    """Class for all entity data in a single prometheus metric."""
+
+    metric_name: str
+    domain: str
+    friendly_name: str
+    object_id: str
+    metric_value: Any | None = None
+    mode: str | None = None
+    state: str | None = None
+    action: str | None = None
+
+    @classmethod
+    def from_entity_id(cls, entity_id: str, **kwargs) -> Self:
+        """Construct EntityTestMetricInfo using full entity_id instead of parts."""
+        domain, object_id = entity_id.split(".")
+        return cls(
+            metric_name=kwargs["metric_name"],
+            domain=domain,
+            friendly_name=kwargs["friendly_name"],
+            object_id=object_id,
+            metric_value=kwargs.get("metric_value"),
+            mode=kwargs.get("mode"),
+            state=kwargs.get("state"),
+            action=kwargs.get("action"),
+        )
+
+    @property
+    def entity(self) -> str:
+        """Generate entity_id from components."""
+        return f"{self.domain}.{self.object_id}"
```

Comment:
I am not sure that encoding label transformations in the testing helper is a good idea. The goal of the test is to confirm that the metric content matches expectations, and it's better to have the test be explicit (even if verbose) rather than hard-code additional logic in the test.

If we expect the metric to have an `entity` label, we should probably pass an `entity` value to these helpers.

Reply:
I rewrote as you suggested. Considering that we already pass in `object_id` and `domain` to pretty much every test, also including `entity` means a lot of repetition. But is rewritten as you suggested.